### PR TITLE
SQ-831: Add new project role 'QUERY'

### PR DESCRIPTION
**Pull request by RMCrean**
This idea originally came from the test user group feedback. Basically: have 1 more `ProjectRoles` that sits in between READ and EDIT roles, named QUERY. The QUERY role can upload files and run queries and run dimensions update. It cannot rm files or rm/update project versions unlike the EDIT role.

The permissions for each role is fully described on the `docs/user-guides/account-management.md` page and summarized on the website's project detail page (with a link to docs site) and in the modal when a project manager adds a new member and set the new members permissions.  

### Summary of changes:
- updates api routes guards accordingly to match the new permissions. 
- Frontend forms etc.. include the new role and display the query role both when managing members and when setting the scope of a PAT. 
- updates docs to include query role and the guide on what each role can do.  
- Added some e2e tests to help validate what each role can do, and also modified some existing tests to use the minimum role that can possibly do the task as a way to improve the validation (e.g. if the tests is for uploading a file, let a query user do it, not an edit user).  
- I modified some of the `AuthorizationError()` messages raised in the api routes to (1) remove "project does not exist" as part of the error message because user must be a member of the project in order to reach that stage otherwise the `Depends(get_project_member)` would have already raised (and said either project does not exist or you are not a member). (2) I also included what role would be needed to do the operation as perhaps useful (didn't do it for READ required operations though as that is the min level). 

### Note on the db migration:
As enums are little special in postgres, the migration script essentially (1) stringyfies the roles, deletes the enum and then recreates the new enum and then casts the strings back into the enum. This worked nicely in local dev (both up and downgrade with a query user created before doing the downgrade). 
I will test this migration on the dev cluster to be sure though. 

### Plan:
- Review from the :robot: 
- test deploy on dev cluster
- open for review

### CODE REVIEW

File: docs/user-guides/account-management.md [hunk 1]
```diff
@@ -11,20 +11,22 @@ Once your project is created, you will be added as a member with the **Manage**
...
-| `version ls` | | ✓ | ✓ |
-| `version add` / `version rm` | | ✓ | ✓ |
-| `task-history project` | | | ✓ |
-| Add / remove project members <sup>1</sup>| | | ✓ |
+Each project member is assigned one of four roles. The table below shows what each role can do within the project using `divbase-cli`.
+
+| CLI Command | Read | Query | Edit | Manage |
+|---|:---:|:---:|:---:|:---:|
+| `files ls` | ✓ | ✓ | ✓ | ✓ |
+| `files download` | ✓ | ✓ | ✓ | ✓ |
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply docs/user-guides/account-management.md:19]: I think implication is that if you can download, then you can download-all, ignoring.

File: packages/divbase-api/src/divbase_api/templates/components/project_members.html [hunk 2]
```diff
@@ -57,10 +57,8 @@ <h5 class="mb-0">
                                 {% if project.user_role == 'manage' %}
                                 <td class="text-center">
                                     {% if member.user_id == current_user.id %}
-                                    <!-- Manager can't edit their own role -->
-                                    <span class="text-muted small">
-                                        <i class="bi bi-shield-check me-1"></i>Owner
-                                    </span>
+                                    <!-- can't modify your own role -->
+                                    <span class="text-muted small"></i>N/A</span>
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply packages/divbase-api/src/divbase_api/templates/components/project_members.html]: good spot, commit 7971e173ba9568db124b506275656881149fce4a

File: packages/divbase-api/src/divbase_api/templates/components/add_user_to_project.html [hunk 3]
```diff
@@ -29,17 +29,22 @@ <h5 class="modal-title" id="addUserToProjectModalLabel">
...
+                            <small class="text-muted">Higher roles inherit all permissions from lower roles</small>
                         </label>
                         <select class="form-select" id="project_role" name="role" required>
                             <option value="">Select a role...</option>
                             <option value="read">
-                                <i class="bi bi-eye"></i> Read - Can view and download files
+                                <i class="bi bi-eye"></i> 1. Read - Can view and download files
+                            </option>
+                            <option value="query">
+                                <i class="bi bi-search"></i> 2. Query - Can run queries and upload files.
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply packages/divbase-api/src/divbase_api/templates/components/add_user_to_project.html]: good point, commit 7971e173ba9568db124b506275656881149fce4a

File: packages/divbase-api/src/divbase_api/templates/project_pages/detail.html [hunk 4]
```diff
@@ -82,6 +87,31 @@ <h5 class="mb-0"><i class="bi bi-terminal me-2"></i>Add this project to your Div
...
+                           class="small text-muted flex-shrink-0 ms-3" target="_blank" rel="noopener">
+                            <i class="bi bi-box-arrow-up-right me-1"></i>Full breakdown of each role
+                        </a>
+                    </div>
+                    <div class="card-body">
+                        <ul class="list-unstyled d-flex flex-wrap gap-3 mb-0">
+                            <li><span class="badge scilife-role-badge-read me-1">Read</span><small class="text-muted">View and download files</small></li>
+                            <li><span class="badge scilife-role-badge-query me-1">Query</span><small class="text-muted">Upload files, submit queries, and add versions</small></li>
+                            <li><span class="badge scilife-role-badge-edit me-1">Edit</span><small class="text-muted">delete files and remove versions</small></li>
+                            <li><span class="badge scilife-role-badge-manage me-1">Manage</span><small class="text-muted">add and remove project members</small></li>
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply packages/divbase-api/src/divbase_api/templates/project_pages/detail.html]: commit 7971e173ba9568db124b506275656881149fce4a

File: docs/user-guides/tutorial-query-on-public-data.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Tutorial: Running a DivBase query on a public dataset
 
-This tutorial assumes that you have an account on DivBase and have a membership in a DivBase project with at least an EDIT role (i.e. can upload files and run queries). If this is your first time using DivBase, you may want to have a look at the [Quick Start guide](quick-start.md) before trying the tutorial out.
+This tutorial assumes that you have an account on DivBase and have a membership in a DivBase project with at least one 'QUERY' level role (i.e. can upload files and run queries). If this is your first time using DivBase, you may want to have a look at the [Quick Start guide](quick-start.md) before trying the tutorial out.
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply docs/user-guides/tutorial-query-on-public-data.md]: commit 7971e173ba9568db124b506275656881149fce4a

File: packages/divbase-api/src/divbase_api/migrations/versions/2026-06-09_add_query_as_project_role.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+revision: str = "9885301b9073"
+down_revision: Union[str, Sequence[str], None] = "427a022ebccf"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    """Upgrade schema."""
+    # manually created steps to add the QUERY role to the projectroles enum
+    # Convert the colum to string first, so we can drop and then recreate the new enum type safely.
```
[the remark this answers is not in the document]
Comment (RMCrean) [reply packages/divbase-api/src/divbase_api/migrations/versions/2026-06-09_add_query_as_project_role.py]: commit 7971e173ba9568db124b506275656881149fce4a

**Comment by RMCrean** [comment]
(Migration on dev cluster instance worked nicely)

**Comment by RMCrean** [comment]
merging now based on internal plan
